File: chat-example/src/main/java/com/hoten/servingchat/client/ConnectionToChatServerHandler.java
package com.hoten.servingchat.client;

import com.hoten.servingjava.SocketHandler;
import com.hoten.servingjava.SocketHandlerImpl;
import com.hoten.servingjava.filetransferring.FileTransferringSocketReciever;
import com.hoten.servingjava.message.JsonMessageBuilder;
import com.hoten.servingjava.message.Message;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.File;
import java.io.IOException;
import java.net.Socket;

public class ConnectionToChatServerHandler implements SocketHandler {

    private final FileTransferringSocketReciever _socketHandler;
    private final Chat _chat;

    public ConnectionToChatServerHandler(Socket socket, Chat chat) throws IOException {
        _socketHandler = new FileTransferringSocketReciever(new SocketHandlerImpl(socket));
        _chat = chat;
    }

    public void sendUsername(String username) throws IOException {
        Message message = new JsonMessageBuilder()
                .type("SetUsername")
                .set("username", username)
                .build();
        send(message);
    }

    public void quit() throws IOException {
        Message message = new JsonMessageBuilder()
                .type("LogOff")
                .build();
        send(message);
    }

    public void sendMessage(String msg) throws IOException {
        Message message = new JsonMessageBuilder()
                .type("ChatMessage")
                /*.compressed(true)*/ // :(
                .set("msg", msg)
                .build();
        send(message);
    }

    public void sendWhisper(String to, String msg) throws IOException {
        Message message = new JsonMessageBuilder()
                .type("Whisper")
                .set("to", to)
                .set("msg", msg)
                .build();
        send(message);
    }

    @Override
    public void start(Runnable onConnectionSettled, SocketHandler topLevelSocketHandler) throws IOException, InstantiationException, IllegalAccessException {
        _socketHandler.start(onConnectionSettled, topLevelSocketHandler);
    }

    @Override
    public void send(Message message) throws IOException {
        _socketHandler.send(message);
    }

    @Override
    public void close() {
        _socketHandler.close();
    }

    @Override
    public DataOutputStream getOutputStream() {
        return _socketHandler.getOutputStream();
    }

    @Override
    public DataInputStream getInputStream() {
        return _socketHandler.getInputStream();
    }

    public File getLocalDataFolder() {
        return _socketHandler.getLocalDataFolder();
    }

    public Chat getChat() {
        return _chat;
    }
}

File: chat-example/src/main/csharp/ChatClient/ChatClient/ConnectionToChatServerHandler.cs
using Serving;
using Serving.FileTransferring;
using System;

namespace ChatClient
{
    class ConnectionToChatServerHandler : SocketHandler
    {
        public Chat Chat { get; private set; }

        private FileTransferringSocketReciever _socketHandler;

        public ConnectionToChatServerHandler(String host, int port, Chat chat)
        {
            _socketHandler = new FileTransferringSocketReciever(new SocketHandlerImpl(host, port));
            Chat = chat;
        }

        public void SendUsername(String username)
        {
            var message = new JsonMessageBuilder()
                    .Type("SetUsername")
                    .Set("username", username)
                    .Build();
            Send(message);
        }

        public void Quit()
        {
            var message = new JsonMessageBuilder()
                    .Type("LogOff")
                    .Build();
            Send(message);
        }

        public void SendMessage(String msg)
        {
            var message = new JsonMessageBuilder()
                    .Type("ChatMessage")
                    /*.Compressed(true)*/ // :(
                    .Set("msg", msg)
                    .Build();
            Send(message);
        }

        public void SendWhisper(String to, String msg)
        {
            var message = new JsonMessageBuilder()
                    .Type("Whisper")
                    .Set("to", to)
                    .Set("msg", msg)
                    .Build();
            Send(message);
        }

        public void Start(Action onConnectionSettled, SocketHandler topLevelSocketHandler)
        {
            _socketHandler.Start(onConnectionSettled, topLevelSocketHandler);
        }
        
        public void Send(Message message)
        {
            _socketHandler.Send(message);
        }

        public void Close()
        {
            _socketHandler.Close();
        }

        public JavaBinaryReader GetInputStream()
        {
            return _socketHandler.GetInputStream();
        }

        public JavaBinaryWriter GetOutputStream()
        {
            return _socketHandler.GetOutputStream();
        }

        public String GetLocalDataFolder()
        {
            return _socketHandler.LocalDataFolder;
        }
    }
}
